src/ct_selftest/ct_selftest.cpp

```diff
@@ -209,6 +209,61 @@ void test_scoped_poison_outer(Botan::RandomNumberGenerator& rng) {
    }
 }
 
+/**
+ * @brief Test that the return value of poison/unpoison is correctly handled.
+ *
+ * In theory we only test unpoison here. Since the logic of poison and unpoison is
+ * equal we expect poison to work as well.
+ */
+void test_unpoison_inout(Botan::RandomNumberGenerator& rng) {
+   const uint8_t poisoned_byte = rng.next_byte();
+   Botan::CT::poison(poisoned_byte);
+
+   // Test that the return value is unpoisoned
+   if(Botan::CT::unpoison(poisoned_byte) == 0) {
+      std::cout << "I may or may not be printed." << std::endl;
+   }
+
+   // Test that the value is unpoisoned
+   if(poisoned_byte == 1) {
+      std::cout << "I may or may not be printed too." << std::endl;
+   }
+
+   auto poisoned_range = rng.random_vec(8);
+   Botan::CT::poison_range(poisoned_range);
+
+   // Test that the return value is unpoisoned
+   if(Botan::CT::unpoison_range(poisoned_range)[0] == 0) {
+      std::cout << "I may or may not be printed." << std::endl;
+   }
+
+   // Test that the value is unpoisoned
+   if(poisoned_range[0] == 0) {
+      std::cout << "I may or may not be printed too." << std::endl;
+   }
+
+   struct Poisanable {
+         uint8_t a;
+
+         void _const_time_poison() { Botan::CT::poison(a); }
+
+         void _const_time_unpoison() { Botan::CT::unpoison(a); }
+   };
```

Comment:
I deliberately didn't test the `_const_time_poison()` overload in `ct_selftest.cpp` (instead its tested in `test_ct_utils.cpp`). After all these helper methods allow to establish an _observable side effect_ that can be nicely tested in a non-valgrind-enabled unit test.

What you're actually testing below is the proper functioning of the _implementation_ of `const_time_poison()`; and you already did test `poison(uint8_t)` above.

Reply:
Well, I wanted to test that the parameters are correctly passed through. To create this struct was simply a tool to test the pass-through of `poison(class)`.